acstools/acszpt.py

```diff
@@ -0,0 +1,329 @@
+            print('-'*79)
+            valid_detector = False
+
+        # Determine if the submitted filter is valid
+        if self.filt is None:
+            pass
+        elif self.filt not in self.valid_filters[self.detector]:
+            valid_filter = False
+            print('Whoops! {} is not a valid filter for {}'.
+                  format(self.filt, self.detector))
+            print('Please choose one of the following:')
+            for filt in self.valid_filters[self.detector]:
+                print(filt)
+            print('-' * 79)
+
+        # Determine if the submitted date is valid
+        date_check = self._check_date()
+        if isinstance(date_check, ValueError):
+            valid_date = False
+            print('Whoops! {}'.format(date_check))
+
+        # Only continute if all the inputs are valid options
+        if all([valid_date, valid_filter, valid_detector]):
+            return True
+        else:
+            return False
+
+    def _check_date(self, fmt='%Y-%m-%d'):
+        """Convenience method for determing if the input date is valid
+        """
+        min_date = dt.datetime(2002, 3, 7)
+        try:
+            dt_obj = dt.datetime.strptime(self.date, fmt)
+        except ValueError:
+            return ValueError('Date format does not match YYYY-MM-DD')
+        else:
+            if min_date < dt_obj:
+                result = None
+            else:
+                result = ValueError('The observation cannot occur before '
+                                 'ACS was installed ({})'.
+                                 format(min_date.strftime(fmt)))
+            return result
+
+    def _submit_request(self):
+        """ Submit a request to the ACS Zeropoint Calculator
+
+        If an exception is raised during the request an error message is
+        printed. Otherwise, the response is saved in the corresponding
+        attribute.
+        """
+        try:
+            self._response = requests.get(self._url)
+        except requests.exceptions.RequestException as e:
+            print(e)
+            print('-'*79)
+            print('The query failed! Check your input args and if the error '
+                  'persists submit a ticket to the ACS Help Desk at '
+                  'hsthelp.stsci.edu with the error message displayed above.')
+            self._failed = True
+
+    def _parse_response(self):
+        """ Parse the html response returned from the HTML request to find all
+        tables on the webpage.
+
+        Using bs4, find all the <tb> </tb> tags present in the response. Save
+        the results in an OrderedDict() so when parsing occurs we get stable
+        results.
+
+        """
+        soup = BeautifulSoup(
+            self._response.content.decode(self._response.encoding),
+            'html.parser'
+        )
+        # Grab all elements in the table returned by the ZPT calc.
+        td = soup.find_all('td')
+        for j in range(6):
```

Comment:
Magic number! Why 6?

Reply:
There are 6 columns in the parsed html table, so I have to loop through in blocks of 6. Couldn't really come up with a cleaner way to do this that didn't involve hard coding that value.